src/main/webapp/query-control.js

```diff
@@ -35,13 +35,82 @@ function clearPreviousResult() {
+function getFetchUrl(servlet, personType, action, location, year) {
+  return '/' + servlet + '?person-type=' + personType +
     '&action=' + action +
     '&location=' + location +
     '&year=' + year;
 }
 
+function getTitle(personType, location, year, locationInput, actionInput) {
+  const isCountyQuery = location !== 'state';
+  const region = isCountyQuery ? locationInput + ' county' : 'U.S. state';
+  const title = 'Population who ' + actionInput +
+    ' each ' + region + ' (' +
+    personType.replace('-', ' ') + ')' +
+    ' in ' + year;
+  return title;
+}
+
+// getHistory fetches to the History Servlet and returns a list of 
+// links of queries that the logged in user has visited before
+function getHistory(personType, action, location, year, userId) {
+  let fetchUrl = getFetchUrl('history', personType, action, location, year);
+  fetchUrl = fetchUrl + '&user-id=' + userId;
+  
+  // clear previous results
+  const historyContainer = document.getElementById('history');
+  historyContainer.innerHTML = '';
+  const header = document.createElement('p');
+  header.innerText = "Pages you've Viewed";
+  historyContainer.appendChild(header);
+
+  // change back the action to be gramatically correct in the title
+  const gramaticallyCorrectAction = new Map();
+  gramaticallyCorrectAction.set(
+        'live', 'lived in',
+      ).set(
+        'work', 'worked in',
+      ).set(
+        'moved', 'moved to',
+      );
+  fetch(fetchUrl).then((response) => response.json()).then(
+    function (jsonResponse) {
+      jsonResponse.forEach((historyElement) => {
+        if (historyElement !== null) {
+          let historyTextNode;
+          if (historyElement.location !== 'state') {
+            const historyText = getTitle(historyElement.personType, historyElement.location, 
+              historyElement.year, stateInfo[historyElement.location].name, 
+              gramaticallyCorrectAction.get(historyElement.action));
+            historyTextNode = document.createTextNode(historyText);
+          } else {
+            const historyText = getTitle(historyElement.personType, historyElement.location, 
+              historyElement.year, 'Each U.S. state', 
+              gramaticallyCorrectAction.get(historyElement.action));
+            historyTextNode = document.createTextNode(historyText);
+          }
+          // Link the titles to the query url
+          const linkElement = document.createElement('a');
+          linkElement.href = getHistoryUrl(historyElement);
+          linkElement.appendChild(historyTextNode);
+          historyContainer.appendChild(linkElement);
+          const breakElement = document.createElement('br');
+          historyContainer.appendChild(breakElement);
+        }
+      })});
+}
+
+// Given a history element, return the appropriate url
+function getHistoryUrl(historyElement) {
+  const host = window.location.origin;
+  const hash = '/#person-type=' + historyElement.personType + 
```

Comment:
You could use fetchUrl here by either modifying fetchUrl or calling fetchUrl as is and then removing the servlet part

Reply:
Good point! I did the latter